src/python/pybind11/core.cc

```diff
@@ -310,7 +310,10 @@ pybind_init_core(py::module_ &m_native)
         .def("disableAllListeners", &ListenSocket::disableAll)
         .def("listenersDisabled", &ListenSocket::allDisabled)
         .def("listenersLoopbackOnly", &ListenSocket::loopbackOnly)
-        .def("seedRandom", [](uint64_t seed) { random_mt.init(seed); })
+        .def("seedRandom", [](uint64_t seed) {
+            for (auto rng_ptr : Random::instances)
+                rng_ptr->init(seed);
```

Comment:
This I believe will only catch cases where the Random is constructed before the seedRandom call. What happens if we call seedRandom before constructing a Random object? This might happen for example with SimObjects.

I believe the logic should be something like

         .def("seedRandom", [](uint64_t seed) {
            globalSeed = seed;
            // Reseed previously constructed instances
            for (auto rng_ptr : Random::instances)
                rng_ptr->init(seed);
          }

With globalSeed defined instead of mt_random as

`uint64_t globalSeed = 5489;`

And the Random constructor being:
```

Random::Random()
{
    init(globalSeed);
}
```

Reply:
I correct myself; rather than a global variable globalSeed could be a static member of Random